tests/conftest.py

```diff
@@ -0,0 +1,187 @@
+'conftest.py' for any unit test executed from within this directory
+tree. This file is used to define fixtures, hooks, plugins, and other
+functionality that can be shared by the unit tests.
+
+This file has been created for the OpenML testing to primarily make use
+of the pytest hooks 'pytest_sessionstart' and 'pytest_sessionfinish',
+which are being used for managing the deletion of local and remote files
+created by the unit tests, run across more than one process.
+
+This design allows one to comment or remove the conftest.py file to
+disable file deletions, without editing any of the test case files.
+
+
+Possible Future: class TestBase from openml/testing.py can be included
+    under this file and there would not be any requirements to import
+    testing.py in each of the unit test modules.
+'''
+
+import os
+import logging
+from typing import List
+
+import openml
+from openml.testing import TestBase
+
+# creating logger for unit test file deletion status
+logger = logging.getLogger("unit_tests")
+logger.setLevel(logging.DEBUG)
+fh = logging.FileHandler('UnitTestDeletion.log')
+logger.addHandler(fh)
+
+file_list = []
+directory = None
+
+# finding the root directory of conftest.py and going up to OpenML main directory
+# exploiting the fact that conftest.py always resides in the root directory for tests
+static_dir = os.path.dirname(os.path.abspath(__file__))
+logging.info("static directory: {}".format(static_dir))
+print("static directory: {}".format(static_dir))
+while True:
+    if 'openml' in os.listdir(static_dir):
+        break
+    static_dir = os.path.join(static_dir, '..')
+
+
+def worker_id() -> str:
+    ''' Returns the name of the worker process owning this function call.
+
+    :return: str
+        Possible outputs from the set of {'master', 'gw0', 'gw1', ..., 'gw(n-1)'}
+        where n is the number of workers being used by pytest-xdist
+    '''
+    vars_ = list(os.environ.keys())
+    if 'PYTEST_XDIST_WORKER' in vars_ or 'PYTEST_XDIST_WORKER_COUNT' in vars_:
+        return os.environ['PYTEST_XDIST_WORKER']
+    else:
+        return 'master'
+
+
+def read_file_list() -> List[str]:
+    '''Returns a list of paths to all files that currently exist in 'openml/tests/files/'
+
+    :return: List[str]
+    '''
+    directory = os.path.join(static_dir, 'tests/files/')
+    if worker_id() == 'master':
+        logger.info("Collecting file lists from: {}".format(directory))
+    files = os.walk(directory)
+    file_list = []
+    for root, _, filenames in files:
+        for filename in filenames:
+            file_list.append(os.path.join(root, filename))
+    return file_list
+
+
+def compare_delete_files(old_list, new_list) -> None:
+    '''Deletes files that are there in the new_list but not in the old_list
+
+    :param old_list: List[str]
+    :param new_list: List[str]
+    :return: None
+    '''
+    file_list = list(set(new_list) - set(old_list))
+    for file in file_list:
+        os.remove(file)
```

Comment:
I'm not sure if that's desired behavior. The unit tests shouldn't leave any new files. Do you know the reason why there are some files left over?

Reply:
This goes back to the original issue we were trying to address. Some files remained on git status after the unit tests were run.
If I remember correctly, some of the tests (especially in the task module) were downloading data, which were shared across unit tests. Also, the tearDown is not defined for a few of these test modules. Whereas the design of the tearDown would I believe demand the manual bookkeeping of all IDs downloaded. 
All in all, it looked like a more robust check to collect and delete all generated files from the 'master' worker.